Python/fitnesse/html_util.py

```diff
@@ -2,41 +2,41 @@
 
 from fitnesse.context import SuiteResponder, PageCrawlerImpl, PageData, PathParser, WikiPage, WikiPagePath
 
-
+class TestType:
+    SETUP: "SetUp"
+    TEARDOWN: "TearDown"
 class HtmlUtil:
 
+    @staticmethod
+    def get_Path(page_name: str, wiki_page: WikiPage):
+        inherit_Page: WikiPage = page_crawler.get_inherited_page(page_name, wiki_page)
+        if inherit_Page is not None:
+            path: wiki_page.get_page_crawler().get_full_path(setup)
+            path_name: str = path_parser.render(page_path)
+            return path_name
+
     @staticmethod
     def testable_html(page_data: PageData, include_suite_setup: bool, page_crawler: PageCrawlerImpl, path_parser: PathParser) -> str:
         wiki_page: WikiPage = page_data.get_wiki_page()
         string_io: StringIO = StringIO()
 
         if page_data.has_attribute("Test"):
```

Comment:
You could calculate `page_data.has_attribute("Test")` on top of the function body, or in a separate function once, as the boolean information is needed a second time (cf. line 32).